File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/AwardQueryImpl.java
package dev.jefy.connectpro.management.appliacaion.query;

import org.jspecify.annotations.NullMarked;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

import dev.jefy.connectpro.management.appliacaion.dtos.AwardResponse;
import dev.jefy.connectpro.management.domain.repositoty.AwardRepository;
import dev.jefy.connectpro.management.domain.vo.AwardId;
import dev.jefy.connectpro.shared.application.exceptions.ResourceNotFound;
import lombok.RequiredArgsConstructor;

/**
 * @author Jôph Yamba
 */
@NullMarked
@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class AwardQueryImpl implements AwardQuery {
    private final AwardRepository awardRepo;
    @Override
    public AwardResponse get(AwardId awardId) {
        return awardRepo
                .findById(awardId)
                .map(AwardResponse::from)
                .orElseThrow(()-> new ResourceNotFound(
                        "Award with id: %s not found".formatted(awardId.value())
                ));
    }

    @Override
    public List<AwardResponse> getAll() {
        return awardRepo
                .findAll()
                .stream()
                .map(AwardResponse::from)
                .toList();
    }
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/AwardQueryImpl.kt
package dev.jefy.connectpro.management.appliacaion.query

import dev.jefy.connectpro.management.appliacaion.dtos.AwardResponse
import dev.jefy.connectpro.management.appliacaion.dtos.toResponse
import dev.jefy.connectpro.management.domain.repository.AwardRepository
import dev.jefy.connectpro.management.domain.vo.AwardId
import dev.jefy.connectpro.portfolio.application.exceptions.AwardNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
 * @author Jôph Yamba
 */
@Service
@Transactional(readOnly = true)
class AwardQueryImpl(private val awardRepo: AwardRepository) : AwardQuery {

    override fun get(awardId: AwardId): AwardResponse = awardRepo
        .findById(awardId)
        .map{ it.toResponse() }
        .orElseThrow { AwardNotFoundException() }
    

    override fun getAll(): MutableList<AwardResponse> = awardRepo
        .findAll()
        .stream()
        .map{ it.toResponse()}
        .toList()
}
